lib/rack/prx_auth/auth_validator.rb

```diff
@@ -0,0 +1,54 @@
+require 'json/jwt'
+require 'net/http'
+
+module Rack
+  class PrxAuth
+    class AuthValidator
+
+      attr_reader :issuer, :claims, :token
+
+      def initialize(token, certificate, issuer)
+        @token = token
+        @certificate = certificate
+        @issuer = issuer
+      end
+
+      def valid?
+        begin
+          decode_token && !expired? && @certificate.valid?(token)
+        rescue JSON::JWT::InvalidFormat
+          false
+        end
+      end
+
+      def claims
+        @claims ||= decode_token
+      end
+
+      def decode_token
+        return {} if token.nil?
+
+        begin
+          JSON::JWT.decode(token, :skip_verification)
+        rescue JSON::JWT::InvalidFormat
+          {}
+        end
+      end
+
+      def expired?
+        return true if claims['iat'].nil? || claims['exp'].nil?
```

Comment:
Is this a meaningful change, or just catching edge cases?  (I don't see this line in the old logic).